aten/src/ATen/native/Copy.cpp

```diff
@@ -320,6 +320,10 @@ void copy_ignoring_overlaps(const TensorBase &dst, const TensorBase &src) {
   copy_stub(iter.device_type(), iter, /*non_blocking=*/false);
 }
 
+void _propagate_xla_data(const Tensor& input, const Tensor& output) {
+  TORCH_INTERNAL_ASSERT(input.device().type() == kXLA, "This op should only be called by XLA")
```

Comment:
Can you actually make this `TORCH_INTERNAL_ASSERT(false, ...)`? The idea is that this kernel should **never** be hit, because if you run with XLA tensors, they'll always hit the XLA implementation (and we should never end up here if we are running with non-xla tensors)